### ISSUE TITLE
Ignore YAML collection in frontmatter of a markdown file

### ISSUE THREAD

**Issue by CyberSinh** [DavidAnson/vscode-markdownlint#283]
Hi,

```markdown
---
title: 'Documentation'
taxonomy:
    category: [documentation, support, docs]
---

**Markdown text**
```

In the code above, the linter complains with message: "No link definition found: 'documentation, support, docs'".
The linter should ignore YAML collection in frontmatter of a markdown file.

Thanks.

**Comment by DavidAnson** [comment]
Replacing both "---" with "\s*---" in that RegExp will probably do what you want.

**Comment by CyberSinh** [comment]
Thanks for your answer. 
But it doesn't work on my side.
According to [regex101.com](https://regex101.com/), the [default regex settings](https://github.com/DavidAnson/markdownlint#optionsfrontmatter)  should already correctly handle white space between `---`.

**Comment by DavidAnson** [comment]
Actually, based on the error message I think this is a duplicate of the following issue: https://github.com/DavidAnson/markdownlint/issues/534

It is a problem with VS Code, not the markdownlint extension.
